pajbot/managers/connection.py

```diff
@@ -51,13 +51,26 @@ def __init__(self, reactor):
         super().__init__(reactor)
 
         self.num_msgs_sent = 0
+        self.num_whispers_sent_minuite = 0
+        self.num_whispers_sent_second = 0
         self.in_channel = False
 
     def reduce_msgs_sent(self):
         self.num_msgs_sent -= 1
 
-    def can_send(self):
-        return self.num_msgs_sent < TMI.message_limit
+    def reduce_whispers_sent_minuite(self):
+        self.num_whispers_sent_minuite -= 1
+
+    def reduce_whispers_sent_second(self):
```

Comment:
these defs should be prefixed with _ since they're private
The fact that previous pajlada or randers didn't do that shouldn't prevent you from following best practices :)

Reply:
They cant be private because they are apart of the Connection class but are referenced in the ConnectionManager class.

Reply:
Ah I see, my bad.